packages/astro/src/core/errors/overlay.ts

```diff
@@ -608,7 +608,10 @@ class ErrorOverlay extends HTMLElement {
 
 		this.text('#name', err.name);
 		this.text('#title', err.title);
-		this.text('#message-content', err.message, true);
+    // Only set as html if escaped. We're only sure it's escaped if name is part
+    // of the error, ie. we transformed it as we wanted.
+    // TODO: capture error at the source and transform it there
```

Comment:
@Princesseuh I tried many things but couldn't find from where the error on initial render is  coming. Basically the flow is as follow:
1. you have failing code: escaped
2. you update your failing code but it's still failing, you save
3. on initial render, it's not escaped (the error has not been enhanced by us yet, where does  it come from??)
4. after 200ms, enhanced error is used (escaped)